tools/build_uwp_app.bat

```diff
@@ -0,0 +1,20 @@
+call VsDevCmd.bat
+
+mkdir ..\build
+cd ..\build
+
+git clone https://github.com/libsdl-org/SDL.git
+git -C SDL checkout b424665e0899769b200231ba943353a5fee1b6b6

```

Comment:
Can we update our SDL dependency and let CMake handle this instead?
https://github.com/diasurgical/devilutionX/blob/master/3rdParty/SDL2/CMakeLists.txt#L17

Reply:
I have to say that I'm not an CMake expert. So I don't know because we need to use a specific project inside their repo (WinRT). So I did it this way so we can build the lib before doing the CMake stuff and then just link the lib itself.

Other than that we need some specific parameters for that WinRT project which is pretty outdated. `/p:PlatformToolset=v143;TargetPlatformVersion=10.0.22000.0;TargetPlatformMinVersion=10.0.14393.0`

Reply:
We can probably make this work by passing an appropriate value of [`-DCMAKE_GENERATOR_TOOLSET=...`](https://cmake.org/cmake/help/latest/variable/CMAKE_GENERATOR_TOOLSET.html#visual-studio-toolset-selection)

This would mean DevilutionX itself will also be compiled with this old toolset but that is probably OK

Reply:
I have tried that and I ended up with a ton off errors related to the windows runtime using wchar* and placing trying to use char* instead.

Reply:
like i said before i'm not a CMake expert so if you find a better way of doing that it would be great but i can only think about 2 options:
1 - keep this as it is
2 - keep a precompiled SDL2 lib in the repo

Reply:
OK, let's keep this as is for now and look into improving it in the future